### ISSUE TITLE
Missing Resource Owner Password Credentials flow

### ISSUE THREAD

**Issue by alex94cp** [zalando/oauth2-client-js#3]
According to the [rfc](https://tools.ietf.org/html/rfc6749#section-1.3.3), the ROPC flow can also be used when there is a high degree of trust between the resource owner and the client, as it's the case with a first-party app, where the use of an implicit flow would hurt UX. OAuth2-client lacks this ROPC flow.

**Comment by prayerslayer** [comment]
Yup, correct. I’ll see when I have some time to implement it.

I’m curious: Do you intend to use the library in the browser or on the server?

**Comment by alex94cp** [comment]
In the browser, it's a simple React app. I plan to go isomorphic in the
future though.

## 

Álex Puchades

**Comment by prayerslayer** [comment]
Aren’t you exposing your client credentials (including the secret) then?

**Comment by alex94cp** [comment]
Nope. OAuth2 distinguishes two client types: confidential and public (see [here](https://tools.ietf.org/html/rfc6749#section-2.1)). Public clients are not required to provide its client_secret (as it's the case for an ajax app). I'm using [this passport strategy](https://github.com/timshadel/passport-oauth2-public-client) on the server to protect the token endpoint.

**Comment by prayerslayer** [comment]
Ah, so you can do the ROPC flow with non-confidential clients as well. I kind of overread that.

**Comment by lasomethingsomething** [comment]
@prayerslayer Do you still want this enhancement? If so, can we open it up to "Help Wanted"?

**Comment by prayerslayer** [comment]
Yes, do want. It's unlikely I get to this myself though, so I added the help label.

**Comment by lasomethingsomething** [comment]
@prayerslayer Great!
